### ISSUE TITLE
Using file.relativePath for BucketObject name results in "duplicate resource arn" error

### ISSUE THREAD

**Issue by sarink** [pulumi/pulumi-synced-folder#8]
### What happened?

When using two synced-folders and two s3 buckets in the same stack, attempting to deploy a file with the same name (eg, `index.html`) in both throws a "duplicate resource" error.

This is because the name being used is `file.relativePath`, rather than `file.fullPath`

### Steps to reproduce

```
new S3BucketFolder('site1', { path: '../site1', bucketName: site1BucketName });
new S3BucketFolder('site2', { path: '../site2', bucketName: site2BucketName });
```

### Expected Behavior

BucketObject names should use the _full_ file path (not the relative file path), so the above code should work

### Actual Behavior

The above code errors with "duplicate resource arn" error

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Hey @sarink thanks for reporting this. Appreciate you filing this and making a proposed fix! 

I looked at the PR quickly and think I have an alternative path that will work more broadly with the way our component model works.

**Comment by cnunciato** [comment]
@AaronFriel Do you want to provide your suggestion for @sarink on the PR?

**Comment by AaronFriel** [comment]
Yeah, I'm working on a more comprehensive PR. Building and testing now.

**Comment by sarink** [comment]
Thanks for the detailed explanation @AaronFriel . Any idea when this will be released?

**Comment by AaronFriel** [comment]
Aiming for a release this week.

**Comment by AaronFriel** [comment]
This was released in v0.10.0 for all languages except the .NET family (C#) which we'll follow up on.
